src/std/io/exception.d

```diff
@@ -73,7 +73,11 @@ unittest
     }
 
     scope e = new IOException(String("Hello, World"));
-    e.toString(&old);
+    version(preview_in)
+        assert(!__traits(compiles, e.toString(&old)));
```

Comment:
Hm... this is a sign something is wrong.

Reply:
With preview in, the compiler decides whether the type is `const scope char[]` or `const scope ref char[]`. Without it, `in` is well defined and it compiles

Reply:
OK, let's just remove this whole part (the `old` function usage). Use the `new_` thing only. Looking back at the reason this was added, it wasn't passing CI at the time, and this was needed to update the compiler (std.io used to be in dmd buildkite).